.buildkite/steps/check-changelog.sh

```diff
@@ -6,5 +6,10 @@ VERSION=$(buildkite-agent meta-data get "agent-version")
 
 if ! grep --quiet "$VERSION" CHANGELOG.md; then
   echo "The CHANGELOG.md is missing an entry for version ${VERSION}"
-  exit 1
+
+  if [[ "${DRY_RUN:-}" == "false" ]] ; then
+    exit 1
+  else
+    echo "Dry Run Mode enabled, so allowing the build to continue"
+  fi
```

Comment:
this change is just for convenience, so dry-run "releases" aren't blocked by a missing changelog